src/core/public/application/scoped_history.ts

```diff
@@ -219,11 +219,26 @@ export class ScopedHistory<HistoryLocationState = unknown>
 
   /**
    * Creates an href (string) to the location.
+   * If `prependBasePath` is true (default), it will prepend the location's path with the scoped history basePath.
    *
    * @param location
+   * @param prependBasePath
    */
-  public createHref = (location: LocationDescriptorObject<HistoryLocationState>): Href => {
+  public createHref = (
+    location: LocationDescriptorObject<HistoryLocationState>,
+    prependBasePath: boolean = true
```

Comment:
nit: could we use an options object for this to make adding things in the future non-breaking?
```suggestion
    { prependBasePath = true }: { prependBasePath?: boolean } = {}
```

Reply:
👍 Don't even know why I didn't do it that way in the first place. Will do.